extensions/common/aws/aws-s3-core/src/main/java/org/eclipse/edc/aws/s3/AwsClientProviderImpl.java

```diff
@@ -58,23 +60,8 @@ public AwsClientProviderImpl(AwsClientProviderConfiguration configuration) {
     }
 
     @Override
-    public S3Client s3Client(String region, SecretToken token) {
-        if (token instanceof AwsTemporarySecretToken) {
-            var temporary = (AwsTemporarySecretToken) token;
-            var credentials = AwsSessionCredentials.create(temporary.getAccessKeyId(), temporary.getSecretAccessKey(), temporary.getSessionToken());
-            return createS3Client(credentials, region);
-        } else if (token instanceof AwsSecretToken) {
-            var secretToken = (AwsSecretToken) token;
-            var credentials = AwsBasicCredentials.create(secretToken.getAccessKeyId(), secretToken.getSecretAccessKey());
-            return createS3Client(credentials, region);
-        } else {
-            throw new EdcException(String.format("SecretToken %s is not supported", token.getClass()));
-        }
-    }
-
-    @Override
-    public S3Client s3Client(String region) {
-        return s3Clients.computeIfAbsent(region, this::createS3Client);
+    public S3Client s3Client(S3ClientRequest s3ClientRequest) {
+        return s3Clients.computeIfAbsent(s3ClientRequest, this::createS3Client);
```

Comment:
the addition to the `s3Client` should be done only if the `SecretToken` is not passed, and there's no need to change the key type from `String` to `S3ClientRequest`